Framework/DataHandling/src/LoadEventNexus.cpp

```diff
@@ -1081,56 +1086,52 @@ void LoadEventNexus::deleteBanks(EventWorkspaceCollection_sptr workspace,
 void LoadEventNexus::createSpectraMapping(
     const std::string &nxsfile, const bool monitorsOnly,
     const std::vector<std::string> &bankNames) {
-  bool spectramap = false;
   m_specMin = getProperty("SpectrumMin");
   m_specMax = getProperty("SpectrumMax");
   m_specList = getProperty("SpectrumList");
 
-  // set up the
   if (!monitorsOnly && !bankNames.empty()) {
-    std::vector<IDetector_const_sptr> allDets;
-
+    const auto &componentInfo = m_ws->getSingleHeldWorkspace()->componentInfo();
+    std::vector<SpectrumDefinition> spectrumDefinitions;
     for (const auto &bankName : bankNames) {
-      // Only build the map for the single bank
-      std::vector<IDetector_const_sptr> dets;
-      m_ws->getInstrument()->getDetectorsInBank(dets, bankName);
+      const auto &bank = m_ws->getInstrument()->getComponentByName(bankName);
```

Comment:
I guess this is a candidate for us adding some kind of component name to `ComponentInfo` as a future step.